### DISCUSSION TITLE
iOS Crash on Local iPhone Device.

### DISCUSSION THREAD

**Discussion by DclearEE** [unoplatform/uno#9470] [Q&A] [answered]
I started a new Uno Platform App project in Visual Studio - Version 17.3.0. I am running Uno - Version 1.5.3.
I decided to try to build, deploy, and debug the Uno App on my physical iPhone 12 Pro. 
The solution was able to build and create an App on my iPhone, Visual Studio then asks me to open the App. When I click the App on my iPhone it opens the purple .NET splash screen and then crashes after a few seconds. I can't get the App to open without crashing.

I am able to run the same Uno app on Android simulators and Windows. 
I have also successfully built and ran iPhone apps with .NET MAUI in Visual Studio and I have set up all the Apple Dev signing requirements to do so.
The Uno App is still on my iPhone as I would expect it to be, but it always crashes on the splash screen..

This is the error I get in VS:
**System.Exception:** 'No class inheriting from a valid Application Delegate found in /var/mobile/Containers/Data/Application/604E51A0-3D6C-4EBF-845F-A2E391EFF219/Documents/UnoDosApp1.Mobile.content/UnoDosApp1.Mobile.dll'

I'm curious if the bug is related to this issue I found on StackOverflow
[https://stackoverflow.com/questions/72976858/xamarin-ios-only-crash-physical-iphone-when-you-open-an-app](https://stackoverflow.com/questions/72976858/xamarin-ios-only-crash-physical-iphone-when-you-open-an-app)

Any ideas at what is happening here?

**Comment by jeromelaban** [comment]
Thanks for the report. This is not a known issue. Is your visual studio connected to your mac and your iPhone is connected to it, or is the iPhone directly connected to your Windows machine?

**Reply by DclearEE** [reply]
Thanks for the response. The iPhone is connected directly to my Windows machine. I know this is a little unconventional but I have gotten it to work before, most recently with the latest build of MAUI.
I can try with my iPhone connected on a Mac in a few hours.

**Reply by jeromelaban** [reply]
Thanks. This particular scenario is very specific and is called "hot restart", and requires manual addition to the app to make it work. We [blogged about this a while back](https://platform.uno/blog/hot-restart-available-in-uno-platform/). Let us know if that helps!

**Reply by jeromelaban** [reply]
Thanks. I see that you're trying to run with .NET 6, and referencing Xamarin.Forms will not work as a result. We do not have updated steps to make this work at this time for iOS. There's likely another trick to include in the app, but it's not yet known.

**Reply by DclearEE** [reply]
It's also not working when I try to build and deploy remotely with my iPhone in a Mac.
Are these issues related to the .NET 6 referencing Xamarin.Forms issue?
![UnoCodeSnip2](https://user-images.githubusercontent.com/16282199/184254519-d77393ab-8835-4733-b26a-4af84b5b5c77.PNG)

**Reply by jeromelaban** [reply]
This is an unknown message. In all cases, because it's a .NET 6 project, you cannot add a reference to Xamarin.Forms. Make sure to delete the bin/obj folders and try deploying again.

**Reply by jonginleee** [reply]
> Thanks. This particular scenario is very specific and is called "hot restart", and requires manual addition to the app to make it work. We [blogged about this a while back](https://platform.uno/blog/hot-restart-available-in-uno-platform/). Let us know if that helps!

@jeromelaban 

I followed the article in the link and followed it, but the problem is still there. I installed the Xamarin Forms package, but it says Namespace doesn't exist. Is there any workaround?

![image](https://user-images.githubusercontent.com/5918681/211964820-18d1ae94-9ee8-4062-8a3a-6d5c83007334.png)

**Reply by Strypper** [reply]
Hi do we have a solution for this ?

**Reply by WenHong-Lau** [reply]
@jeromelaban I'm having the same exception debugging on Uno 5.2 project. I am deploying to a local iPhone from Windows machine as well.

**Reply by jeromelaban** [reply]
@WenHong-Lau can you open another discussion [with binlogs](https://platform.uno/docs/articles/uno-builds-troubleshooting.html)?

**Reply by WenHong-Lau** [reply]
https://github.com/unoplatform/uno/discussions/17621

**Comment by DclearEE** [comment, marked as answer]
This was fixed for me. The issue was on the Mac and Xcode side. 
I'm honestly not sure what solved it. I tried reinstalling XCode, restarting the Mac, and messing around with the certification for Apple development and something eventually gave.
This was not an error with Uno.
